haiku/_src/stateful.py

```diff
@@ -122,12 +122,8 @@ def grad(fun, argnums=0, has_aux=False, holomorphic=False):
                                       holomorphic=holomorphic)
 
   def grad_fn(*args, **kwargs):
-    value, grads = value_and_grad_fun(*args, **kwargs)
-    if has_aux:
-      value, aux = value
-      return grads, aux
-    else:
-      return grads
+    (_, aux), grads = value_and_grad_fun(*args, **kwargs)
```

Comment:
I don't think this is valid in the has_aux=False case.
Does the test suite pass?